### PR TITLE
[DEVOPS-123] Added optional ability to perform supported image versions checks.

### PR DESCRIPTION
**Pull request by barbun**
# Description
Currently, PaaS customers have the ability to use unsupported image versions. This can introduce issues when an unsupported version of an image is being used (eg. PHP 7, when the last supported version is 8).

# Proposed solution
Extend `docker-base-image` check to allow optional version constrain within the `allowed` list, which will reject builds that do not adhere to the specified image version constraint.

If the version constraint is added to the image, then that version will be considered the last supported version, meaning that images with lower version will not pass validation.

For version comparison it is proposed to use https://github.com/hashicorp/go-version that follows [SemVer](http://semver.org/)

# Usage

The version constraint can be added either with `:` or `@` suffix tag. When no version suffix is specified then any version will pass the validation (current behaviour)

```
docker-base-image:
    - name: '[FILE] Allowed image lookup'
      paths:
        - ./
      allowed:
        - govcms/nginx-drupal:9.5.3
        - uselagoon/varnish-drupal@latest

```

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -217,6 +218,38 @@ func StringSliceMatch(slice []string, item string) bool {
 	return false
 }
 
+// Sift through a slice to determine if it contains eligible package
+// with optional version constrains.
+func SliceCheckString(slice []string, item string, item_version string) bool {
```
Comment (steveworley) [inline pkg/utils/utils.go:223]:
```suggestion
func PackageSliceCheck(slice []string, item string, item_version string) bool {
```
`SliceCheckString` is a bit ambiguous if this is going to perform regex matches to validate packages and version constraints, I think we should aim for a function name that is closer to what this is doing.

**Review by steveworley — CHANGES_REQUESTED** [review]
Minor change on the function name so that it better represents what the function is doing.

File: pkg/utils/utils.go (hunk 1 above)
Comment (barbun) [reply pkg/utils/utils.go]: Good idea! Thanks and done!

**Review by steveworley — APPROVED** [review]
LGTM

**Review by yusufhm — APPROVED** [review]
LGTM

**Pull request merged**